packages/expo-updates/.tests/e2e/__tests__/utils/server.ts

```diff
@@ -0,0 +1,43 @@
+import express from 'express';
+
+const app: any = express();
+let server: any;
+
+let notifyString: string | null = null;
+app.get('/notify/:string', (req: any, res: any) => {
+  notifyString = req.params.string;
+  res.set('Cache-Control', 'no-store');
+  res.send('Received request');
+})
+
+export function start(port: number) {
+  if (!server) {
+    server = app.listen(port);
+  }
+}
+
+export function stop() {
+  if (server) {
+    server.close();
+    server = null;
+  }
+}
+
+function delay(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+export async function waitForResponse(timeout: number) {
+  let delayLength = 0;
```

Comment:
Instead of adding up the delays, better to track the starting time to reduce drift since what we care about is how much time has elapsed since the start of this function